pkg/controller/cluster/main-controller.go

```diff
@@ -302,6 +302,16 @@ func (c *Controller) applyOperatorWebhookSecret(ctx context.Context, mi *miniov1
 				Data: map[string][]byte{
 					miniov1.WebhookOperatorUsername: []byte(cred.AccessKey),
 					miniov1.WebhookOperatorPassword: []byte(cred.SecretKey),
+					miniov1.WebhookMinIOArgs: []byte(fmt.Sprintf("%s://%s:%s@%s:%s%s/%s/%s",
+						"env",
+						cred.AccessKey,
+						cred.SecretKey,
+						fmt.Sprintf("operator.minio-operator.svc.%s",
```

Comment:
There is a problem here, actually not wrt this PR, but the original PR itself. This DNS name of hard coded assuming namespace of operator to be `minio-operator`, but this lookup will fail if the operator is deployed in a different namespace. Ref: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#a-aaaa-records

The namespace should be dynamic for it to work properly.

Reply:
Good catch!

Reply:
We can fix this in subsequent service creation PR, that i'll send.

Reply:
It is on purpose that it is hardcoded, because lookup of namespace was kludgy from the operator itself. 

The way you described to use downward API